### ISSUE TITLE
Feature: Allow node pools to be scaled to zero

### ISSUE THREAD

**Issue by mohag** [rancher/rancher#21005]
**What kind of request is this (question/bug/enhancement/feature request):**
Feature request

**Steps to reproduce (least amount of steps as possible):**
Create a node pool on a AWS, scale it to one. Attempt to scale it down to zero.

**Result:**
"-" button is grayed out / disabled.

**Other details that may be helpful:**
Setting the `quantity` field to zero using the API is accepted. Once there are no nodes, the zero-scaled node pool disappears from the UI, even though it is still visible in the API.

Due to #15739, it might be desirable to have less than 3 worker nodes at times. If a group is created per-AZ, this means that to do it from the UI, some of the node pools needs to be deleted for this.

**Environment information**
- Rancher version v2.2.4
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): AWS nodes
- Machine type (cloud/VM/metal) and specifications (CPU/memory): m5.2xlarge
- Kubernetes version (use `kubectl version`): v1.13.5

**Comment by mohag** [comment]
Another use case is pools per-AZ for spot and non-spot instances, where some might make sense to scale to zero depending on the spot price. (Spot fleet support or auto-scaler support would make this more practical though...)

(There are [documentation on setting up EKS/KOPS clusters with autoscalers](https://blog.powerupcloud.com/running-kubernetes-workloads-on-aws-spot-instances-part-viii-953c3b0b446a) to prefer workloads on spot instances, but move them to on-demand instances automatically when the spot instances get terminated. A manual version of this can become possible with this)

**Comment by ekristen** [comment]
Yes please. This would be great.

**Comment by ekristen** [comment]
I'd like to bring attention back to this. I have a few use cases but one of the most prevalent use cases is auto scaling between spot and on-demand instances. 

Another use case I have is that I have some jobs that need to run only once a day that are heavy ETL and it would be nice to have a defined node pool attached to a node template that can be brought up for a few hours and then taken back down by just the scaling feature instead of a full on cluster edit.

**Comment by ekristen** [comment]
@alena1108 or @cloudnautique any chance this could be added? 

There are times where a particular instance type is at capacity in a region and it would be useful for spot instance node pools to be able to be dropped to zero just like any other resources in k8s so avoid errors and trying to spin up spot instances that are unavailable.

**Comment by Negashev** [comment]
Any updates?
